### DISCUSSION TITLE
Auth-detection in loaders

### DISCUSSION THREAD

**Discussion by jotaachm** [remix-run/react-router#10632] [Q&A]
Hi!

I read some other discussions and understood that you need to check auth-status at every loader and that you should read from the source of truth directly. However, for example, in my app, there are some routes that only managers should have access to so I need to check not only if the user is logged in but also if this user is a manager which is stored in their user object. In my context, I have a getUser function that gives me the information I need about the user by fetching from the relevant endpoint in my API. Since I can't access this context in my loaders, does this mean the only other option is to call that endpoint in every loader? This doesn't feel very efficient so I was wondering if anyone had any other solutions.

On a separate note, I'd love some clarification on which loaders run in parallel and why you need to check auth at every loader.

Thanks!

**Comment by ahrbil** [comment]
Hello @jotaachm, any updates on this? Have you figured it out?

**Reply by sergiodxa** [reply]
Call they getUser in every loader, but inside getUser read from an in-memory cache, so you really only fetch once, then cache it

TanStack Query is really nice for that, use the QueryClient to run a query that will be cached on a certain key, the client will also ensure duplicate calls are deduped and batched so if 3 loaders run only once request is made even if it’s not cached.

And you can invalidate the key cache on an action if you need.

**Comment by ncullen93** [comment]
I have the same problem. It would be nice to avoid the duplication of code and redundant fetching without using tanstack query, but it may be unavoidable.
